packages/near-membrane-base/src/membrane.ts

```diff
@@ -705,6 +720,7 @@ export function createMembraneMarshall() {
                     foreignTargetTraits,
                     foreignTargetFunctionName
                 );
+                this.targetTraits = foreignTargetTraits;
```

Comment:
the value of targetTraits.isArray can be used to avoid adding pending traps this time around.